api/client/beacon/client.go

```diff
@@ -71,7 +71,7 @@ func idTemplate(ts string) func(StateOrBlockId) string {
 		b := bytes.NewBuffer(nil)
 		err := t.Execute(b, struct{ Id string }{Id: string(id)})
 		if err != nil {
-			panic(fmt.Sprintf("invalid idTemplate: %s", ts))
+			return fmt.Sprintf("invalid idTemplate: %s", ts)
```

Comment:
I think this shouldn't be changed, it is essentially an init time operation with usage like `var getBlockRootTpl = idTemplate(getBlockRootPath)`

Reply:
I struggled with this one because it's possible that a future author uses this at runtime. I'll look into refactoring this initialization to happen within an init function where this method is scoped to that function only.

```go
var templateA templateFn
var templateB templateFn

func init() {
  fn := /* todo: insert code from idTemplate */
  templateA = fn("a")
  templateB = fn("b")
}
```

Reply:
Fixed